File: clients/javascript/src/resources/validation.ts
/**
 * Validates that a string parameter is non-empty and returns it URL-encoded.
 * Used across all resource modules for path parameter validation.
 */
export function requireId(value: string, paramName = "id"): string {
  if (!value || typeof value !== "string" || !value.trim()) {
    throw new Error(`${paramName} is required and must be a non-empty string`);
  }
  return encodeURIComponent(value.trim());
}

File: clients/python/src/hiero_proxy/resources/validation.py
"""Shared validation helpers."""

from __future__ import annotations

from urllib.parse import quote


def require_id(value: str, param_name: str = "id") -> str:
    """Validate that a string param is non-empty and return it URL-encoded."""
    if not value or not isinstance(value, str) or not value.strip():
        raise ValueError(f"{param_name} is required and must be a non-empty string")
    return quote(value.strip(), safe="")
